backend/shared/rest/src/main/java/com/meemas/shared/BooleanExpression.java

```diff
@@ -0,0 +1,16 @@
+package com.meemas.shared;
+
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+public class BooleanExpression extends Expression {
+  public BooleanOp op;
```

Comment:
```suggestion
 BooleanOp op;
```

Should not be public -- lombok will generate setters